src/system_identity/identity_creator.rs

```diff
@@ -0,0 +1,50 @@
+use super::creation_response::SystemIdentityCreationResponseData;
+
+/// Interface describing being able to create L2 System Identities.
+pub trait L1IdentityCreator {
+    // TODO type Output;
+    type Error: std::error::Error;
+    fn create_l1_system_identity(&self) -> Result<SystemIdentityCreationResponseData, Self::Error>;
```

Comment:
The output for L1 and L2 is different. Could we create the types now?

Reply:
Added!